plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/gitlabMergeRequest.ts

```diff
@@ -0,0 +1,197 @@
+            type: 'string',
+            title: 'Repository Location',
+            description: `Accepts the format 'gitlab.com/group_name/project_name' where 'project_name' is the repository name and 'group_name' is a group or username`,
+          },
+          projectid: {
+            type: 'string',
+            title: 'projectid',
+            description: 'Project ID of the Gitlab Project',
+          },
+          title: {
+            type: 'string',
+            title: 'Merge Request Name',
+            description: 'The name for the merge request',
+          },
+          description: {
+            type: 'string',
+            title: 'Merge Request Description',
+            description: 'The description of the merge request',
+          },
+          destinationBranch: {
+            type: 'string',
+            title: 'Destination Branch name',
+            description: 'The description of the merge request',
+          },
+          targetPath: {
+            type: 'string',
+            title: 'Repository Subdirectory',
+            description: 'Subdirectory of repository to apply changes to',
+          },
+        },
+      },
+      output: {
+        type: 'object',
+        properties: {
+          projectid: {
+            title: 'Gitlab Project id',
+            type: 'string',
+          },
+          mergeRequestURL: {
+            title: 'MergeRequest(MR) URL',
+            type: 'string',
+            description: 'Link to the merge request in GitLab',
+          },
+        },
+      },
+    },
+    async handler(ctx) {
+      const repoUrl = ctx.input.repoUrl;
+      const { host } = parseRepoUrl(repoUrl, integrations);
+      const integrationConfig = integrations.gitlab.byHost(host);
+
+      const actions: CommitAction[] = [];
+      const formatedTimestamp = () => {
+        const d = new Date();
+        const date = d.toISOString().split('T')[0];
+        const time = d.toTimeString().split(' ')[0].replace(/:/g, '_');
+        return `${date}_${time}`;
+      };
+      const destinationBranch = ctx.input.destinationBranch
+        ? ctx.input.destinationBranch + formatedTimestamp()
+        : `backstage_${formatedTimestamp()}`;
+
+      if (!integrationConfig) {
+        throw new InputError(
+          `No matching integration configuration for host ${host}, please check your integrations config`,
+        );
+      }
+
+      if (!integrationConfig.config.token) {
+        throw new InputError(`No token available for host ${host}`);
+      }
+
+      const api = new Gitlab({
+        host: integrationConfig.config.baseUrl,
+        token: integrationConfig.config.token,
+      });
+
+      const fileRoot = ctx.workspacePath;
+      const localFilePaths = await globby([`${ctx.input.targetPath}'/**'`], {
+        cwd: fileRoot,
+        gitignore: true,
+        dot: true,
+      });
+
+      const fileContents = await Promise.all(
+        localFilePaths.map(p => readFile(path.resolve(fileRoot, p))),
```

Comment:
Let's switch the `path.resolve` our for a `resolveSafeChildPath` here, just in case